lib/launcher.js

```diff
@@ -270,7 +270,13 @@ class Launcher {
             return
         }
         this.logBuffer.add({ level: 'system', msg: 'Starting Node-RED' })
-        const appEnv = Object.assign({}, this.env, this.settings.env)
+        const filter_env = (env) =>
+            Object.entries(env).reduce((acc, [key, value]) =>
+                key.startsWith('FORGE') ? acc : {...acc, [key]: value}, {});
+        const appEnv = Object.assign({},
+            this.settings.env.FORGE_ALLOW_MANAGED_ENV_IN_EDITOR ? filter_env(process.env) : {},
```

Comment:
```suggestion
            this.settings.env.FORGE_EXPOSE_HOST_ENV ? filter_env(process.env) : {},
```

I've proposed a slight rename for this env var to, in my opinion, better describe its purpose.